#### Import Neccessary Libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb
import json
%matplotlib inline
pd.set_option('display.max_colwidth', None)

#### About Data

The <b> Health </b> dataset includes related indicators for each region.

#### Schema

**smoking_prevalence** - Percentage of smokers in population

**diabetes_prevalence** - Percentage of persons with diabetes in population

**infant_mortality_rate** - Infant mortality rate (per 1,000 live births)

**adult_male_mortality_rate** - Mortality rate, adult, male (per 1,000 male adults)

**adult_female_mortality_rate** - Mortality rate, adult, female (per 1,000 male adults)

**pollution_mortality_rate** - Mortality rate attributed to household and ambient air pollution, age-standardized (per 100,000 population)

**comorbidity_mortality_rate** - Mortality from cardiovascular disease, cancer, diabetes or cardiorespiratory disease between exact ages 30 and 70

**hospital_beds** - Hospital beds (per 1,000 people)

**nurses** - Nurses and midwives (per 1,000 people)

**physicians** - Physicians (per 1,000 people)

**health_expenditure** - Health expenditure per capita

**out_of_pocket_health_expenditure** - Out-of-pocket health expenditure per capita


#### Read Data

To prevent Pandas from recognizing the string "NA" (which represents the country Namibia) as a null value, set the argument *keep_default_na* to False and pass a list of values to be interpreted as null (which excludes "NA") to *na_values* when reading the CSV file.

In [5]:
#Exclude "NA" from list of na_values
#na_values = ["","#N/A","#N/A N/A","#NA","-1.#IND","-1.#QNAN","-NaN","-nan","1.#IND","1.#QNAN","<NA>","N/A","NULL","NaN","n/a","nan","null"]

health = pd.read_json('health.json',orient='split')
health

location_key  life_expectancy  smoking_prevalence  diabetes_prevalence  \
0              AD              NaN                33.5                  7.7   
1              AE           77.814                28.9                 16.3   
2              AF           64.486                 NaN                  9.2   
3              AG           76.885                 NaN                 13.1   
4              AL           78.900                28.7                  9.0   
...           ...              ...                 ...                  ...   
3499           WS           73.187                27.7                  9.2   
3500           YE           66.096                18.4                  5.4   
3501           ZA           63.857                20.3                 12.7   
3502           ZM           63.510                13.8                  4.5   
3503           ZW           61.195                15.8                  1.8   

      infant_mortality_rate  adult_male_mortality_rate  \
0                       2.7                        NaN   
1                       6.5                     69.555   
2                      47.9                    237.554   
3                       5.0                    126.917   
4                       7.8                     93.315   
...                     ...                        ...   
3499                   13.6                    138.239   
3500                   42.9                    219.749   
3501                   28.5                    376.404   
3502                   40.4                    321.143   
3503                   33.9                    373.207   

      adult_female_mortality_rate  pollution_mortality_rate  \
0                             NaN                       NaN   
1                          44.863                      54.7   
2                         192.532                     211.1   
3                          83.136                      29.9   
4                          49.486                      68.0   
...                           ...                       ...   
3499                       80.735                      85.0   
3500                      174.558                     194.2   
3501                      248.816                      86.7   
3502                      221.573                     127.2   
3503                      337.702                     133.0   

      comorbidity_mortality_rate  hospital_beds_per_1000  nurses_per_1000  \
0                            NaN                     NaN           4.0128   
1                           16.8                     NaN           5.7271   
2                           29.8                     0.5           0.1755   
3                           22.6                     NaN           4.5171   
4                           17.0                     NaN           3.6495   
...                          ...                     ...              ...   
3499                        20.6                     NaN           2.4885   
3500                        30.6                     NaN           0.7852   
3501                        26.2                     NaN           1.3078   
3502                        17.9                     NaN           1.3376   
3503                        19.3                     NaN           1.9346   

      physicians_per_1000  health_expenditure_usd  \
0                  3.3333             4040.786621   
1                  2.5278             1357.017456   
2                  0.2782               67.122650   
3                  2.9560              673.859680   
4                  1.2164                     NaN   
...                   ...                     ...   
3499               0.3445              233.065063   
3500                  NaN               72.039467   
3501               0.9054              499.237549   
3502               1.1867               67.648666   
3503               0.2096              110.149620   

      out_of_pocket_health_expenditure_u

#### Prepare Data

In [12]:
health = health[health['location_key'].str.contains("_") == False].reset_index(drop=True)
health.head()

location_key  life_expectancy  smoking_prevalence  diabetes_prevalence  \
0           AD              NaN                33.5                  7.7   
1           AE           77.814                28.9                 16.3   
2           AF           64.486                 NaN                  9.2   
3           AG           76.885                 NaN                 13.1   
4           AL           78.900                28.7                  9.0   

   infant_mortality_rate  adult_male_mortality_rate  \
0                    2.7                        NaN   
1                    6.5                     69.555   
2                   47.9                    237.554   
3                    5.0                    126.917   
4                    7.8                     93.315   

   adult_female_mortality_rate  pollution_mortality_rate  \
0                          NaN                       NaN   
1                       44.863                      54.7   
2                      192.532                     211.1   
3                       83.136                      29.9   
4                       49.486                      68.0   

   comorbidity_mortality_rate  hospital_beds_per_1000  nurses_per_1000  \
0                         NaN                     NaN           4.0128   
1                        16.8                     NaN           5.7271   
2                        29.8                     0.5           0.1755   
3                        22.6                     NaN           4.5171   
4                        17.0                     NaN           3.6495   

   physicians_per_1000  health_expenditure_usd  \
0               3.3333             4040.786621   
1               2.5278             1357.017456   
2               0.2782               67.122650   
3               2.9560              673.859680   
4               1.2164                     NaN   

   out_of_pocket_health_expenditure_usd  
0                           1688.121460  
1                            256.034485  
2                             50.665913  
3                            235.749039  
4                                   NaN

#### Data Info

In [13]:
health.shape #size of dataset

(210, 14)

In [14]:
health.columns #column names

Index(['location_key', 'life_expectancy', 'smoking_prevalence',
       'diabetes_prevalence', 'infant_mortality_rate',
       'adult_male_mortality_rate', 'adult_female_mortality_rate',
       'pollution_mortality_rate', 'comorbidity_mortality_rate',
       'hospital_beds_per_1000', 'nurses_per_1000', 'physicians_per_1000',
       'health_expenditure_usd', 'out_of_pocket_health_expenditure_usd'],
      dtype='object')

The health dataset has 210 rows and 14 columns.

#### Check Data Types

In [15]:
health.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 210 entries, 0 to 209
Data columns (total 14 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   location_key                          210 non-null    object 
 1   life_expectancy                       205 non-null    float64
 2   smoking_prevalence                    146 non-null    float64
 3   diabetes_prevalence                   209 non-null    float64
 4   infant_mortality_rate                 193 non-null    float64
 5   adult_male_mortality_rate             189 non-null    float64
 6   adult_female_mortality_rate           189 non-null    float64
 7   pollution_mortality_rate              183 non-null    float64
 8   comorbidity_mortality_rate            183 non-null    float64
 9   hospital_beds_per_1000                25 non-null     float64
 10  nurses_per_1000                       180 non-null    float64
 11  physicians_per_1000

column "location_key" is an object but it should be converted to a string.

In [16]:
health['location_key'] = health['location_key'].astype("string")

In [17]:
health.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 210 entries, 0 to 209
Data columns (total 14 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   location_key                          210 non-null    string 
 1   life_expectancy                       205 non-null    float64
 2   smoking_prevalence                    146 non-null    float64
 3   diabetes_prevalence                   209 non-null    float64
 4   infant_mortality_rate                 193 non-null    float64
 5   adult_male_mortality_rate             189 non-null    float64
 6   adult_female_mortality_rate           189 non-null    float64
 7   pollution_mortality_rate              183 non-null    float64
 8   comorbidity_mortality_rate            183 non-null    float64
 9   hospital_beds_per_1000                25 non-null     float64
 10  nurses_per_1000                       180 non-null    float64
 11  physicians_per_1000

#### Check for Missing Values

In [18]:
missing_values = health.isnull().sum()
missing_values

location_key                              0
life_expectancy                           5
smoking_prevalence                       64
diabetes_prevalence                       1
infant_mortality_rate                    17
adult_male_mortality_rate                21
adult_female_mortality_rate              21
pollution_mortality_rate                 27
comorbidity_mortality_rate               27
hospital_beds_per_1000                  185
nurses_per_1000                          30
physicians_per_1000                      46
health_expenditure_usd                   24
out_of_pocket_health_expenditure_usd     24
dtype: int64

Except for location_key, other columns have missing values ranging from 1 to 185.